addons/udes_common/models/fields.py

```diff
@@ -0,0 +1,59 @@
+import datetime as dt
+from odoo import fields
+
+
+DATE_LENGTH = fields.DATE_LENGTH
+DATETIME_FORMAT = fields.DATETIME_FORMAT + ".%f"
+DATETIME_LENGTH = len(dt.datetime.now().strftime(DATETIME_FORMAT))
+
+
+class PreciseDatetime(fields.Datetime):
+    """Store microsecond-precision timestamps."""
+
+    type = "precise_datetime"
+    column_type = ("timestamp", "timestamp")
+    column_cast_from = ("date",)
+
+    @staticmethod
+    def now(*args):
+        """ Return the current day and time in the format expected by the ORM.
+            This function may be used to compute default values.
+        """
+        return dt.datetime.now().strftime(DATETIME_FORMAT)
+
+    @staticmethod
+    def from_string(value):
```

Comment:
Can we change to `to_datetime`  instead to be consistent with Odoo versioning. In v12 they say they are dropping it, but still there atm :)

Reply:
When i change from_string method to to_datetime it raises error.

Reply:
What error does it raise? Anything useful?